receiver/hostmetricsreceiver/internal/scraper/cpuscraper/internal/metadata/generated_metrics.go

```diff
@@ -24,75 +24,118 @@ import (
 // Type is the component type name.
 const Type config.Type = "cpu"
 
-// MetricIntf is an interface to generically interact with generated metric.
-type MetricIntf interface {
-	Name() string
-	New() pdata.Metric
-	Init(metric pdata.Metric)
+// CpuMetric represents a template for CPU metric creation.
+//
+// Typical use case:
+// m := CpuMetric.Init(startTime)
+// m.EnsureDataPointsCapacity(1)
+// m.Record(now, value, "cpu0", "idle")
+// m.AppendToMetricSlice(metrics)
+type CpuMetric struct {
+	mb             CpuMetricBuilder
```

Comment:
I'm not convinced we need to have both a `CpuMetric` and `CpuMetricBuilder`. It seems the only times we use the builder are 1) for checking how metric is configured, and 2) initializing the metric.

Then in scraper code, there is no new concept of builders. The scraper author does not need to think about builders really, and changes how they instantiate the metric. It's a little weird to have to pass in the `MetricsConfig`, but it might be ok. WDYT?

```go
type CpuMetric struct {
	enabled        bool // later this can be a config struct
	metric         pdata.Metric
	startTimestamp pdata.Timestamp
}

func NewCpuMetric(mc MetricsConfig, startTimestamp pdata.Timestamp) CpuMetric {
	if !mc.SystemCPUTime.Enabled {
		return CpuMetric{...}
	}

	m := pdata.NewMetric()
	m.SetName(mb.Name())
	m.SetDescription(mb.description)
	m.SetUnit(mb.unit)
	m.SetDataType(mb.dataType)
	m.Sum().SetIsMonotonic(mb.isMonotonic)
	m.Sum().SetAggregationTemporality(mb.temporality)
	return CpuMetric{
		enabled: mc.SystemCPUTime.Enabled,
		metric: m, 
		startTimestamp: startTimestamp,
	}
}
```

Reply:
That makes sense. I'll see what can we do